blocks/media-upload-button/index.js

```diff
@@ -77,13 +34,9 @@ class MediaUploadButton extends Component {
 			frameConfig.library = { type };
 		}
 
-		if ( gallery ) {
-			const GalleryDetailsMediaFrame = getGalleryDetailsMediaFrame();
-			this.frame = new GalleryDetailsMediaFrame( {
-				frame: 'select',
-				mimeType: type,
```

Comment:
My thinking here was that passing a custom frame makes the other props redundant, since both audio and video playlists would require their own distinct frames, which would take care of `mimeType` as part of the default props for each frame. However, I would prefer to avoid that confusion if possible, which is what makes the idea of passing a `states` property an interesting one.

Regardless, I think the current implementation of the `MediaUploadButton` is too tightly coupled to the image select workflow, and needs to be reworked a bit to be more generic so we can support additional media workflows.